docs/development/commands.md

```diff
@@ -0,0 +1,137 @@
+
+```java
+@Override
+public void processCommand(ICommandSender sender, String[] args) throws CommandException {
+    // Make sure to check the array length before checking an argument
+    if (args.length == 1 && args[0].equals("confirm")) {
+        LogManager.getLogger("CrashCommand").info("Intentionally crashing the Game!");
+        FMLCommonHandler.instance().exitJava(1, false);
+    } else {
+        sender.addChatMessage(new ChatComponentText("§aAre you sure you want to crash the game? Click to confirm!")
+                .setChatStyle(new ChatStyle()
+                    .setChatClickEvent(new ClickEvent(ClickEvent.Action.RUN_COMMAND, "/crashme confirm"))));
+    }
+}
+```
+
+!!! info
+    Because `sender` is always the current player, you can also use
+    ```java
+    Minecraft.getMinecraft().thePlayer.addChatMessage(/* ... */);
+    ```
+
+Minecraft uses `IChatComponent`s in chat (and a few other places). You can make those by calling `new ChatComponentText("")`. In there you can use format codes like `§a`. If you want, you can also use `EnumChatFormatting.GREEN.toString()` instead of `§a`. You can change the chat style of a `ChatComponentText` in order to give it hover or click effects.
+
+
+!!! warning
+    You might be tempted to open a gui from your command like this:
+    ```java
+    @Override
+    public void processCommand(ICommandSender sender, String[] args) throws CommandException {
+        Minecraft.getMinecraft().displayGuiScreen(new MyGuiScreen());
+    }
+    ```
+    This will not work, since your command gets executed from the chat gui and sending a chat line schedules the chat gui to be closed in the same tick (accidentally closing your gui instead).
+    
+    In order to make this work, you need to instead wait a tick and then open your gui. You can do this by having a tick event handler in your main mod class like this: 
+    ```java
+    // In your main mod class
+    public static GuiScreen screenToOpenNextTick = null;
+
+    @SubscribeEvent
+    public void onTick(TickEvent.ClientTickEvent event) {
+        if (event.phase == TickEvent.Phase.END) return;
+        if (screenToOpenNextTick != null) {
+            Minecraft.getMinecraft().displayGuiScreen(screenToOpenNextTick);
+            screenToOpenNextTick = null;
+        }
+    }
+
+    // In your command class:
+    @Override
+    public void processCommand(ICommandSender sender, String[] args) throws CommandException {
+        ExampleMod.screenToOpenNextTick = new MyGuiScreen();
+    }
+    ```
+
+    See [Events](/development/events) for more info on how to set up event handlers.
+
+## Registering your command
+
+After all this work your command still just will not run. This is because the final step of client commands is still missing. You need to register your command. You typically do this in the `FMLInitializationEvent`:
```

Comment:
```suggestion
After all this work, your command will still not run. This is because the final step of client commands is still missing. You need to register your command. You typically do this in the `FMLInitializationEvent`:
```